### PR TITLE
Update go version to 1.13.7 and add a build stage to the Dockerfile

### PR DESCRIPTION
**Pull request by norrby-s**
* Add a golang:1.13.7 build stage to the Dockerfile
* Update go version in .travis.yml

I think the image build options would be:

Branch: release/v0.3.6
Docker tag: metrics-server-amd64:v0.3.6
Dockerfile location: deploy/docker/Dockerfile
Build context: /

### CODE REVIEW

File: deploy/docker/Dockerfile [hunk 1]
```diff
@@ -1,5 +1,10 @@
-FROM BASEIMAGE
+FROM golang:1.13.7 as builder
+COPY . /go/src/github.com/kubernetes-incubator/metrics-server
+RUN cd /go/src/github.com/kubernetes-incubator/metrics-server && \
+    make _output/amd64/metrics-server
 
-COPY metrics-server /
+FROM gcr.io/distroless/static:latest
```
Comment (johnmccabe) [inline deploy/docker/Dockerfile]: Can you pin to a specific release (sha). Also make sure to use the nonroot variant.

**Review by johnmccabe — CHANGES_REQUESTED** [review]

**Review by johnmccabe — COMMENTED** [review]
How do we trigger hub builds on other forks, are we using tagging?

File: deploy/docker/Dockerfile (hunk 1 above)
Comment (norrby-s) [reply deploy/docker/Dockerfile]: Pinned to latest nonroot sha

**Review by johnmccabe — APPROVED** [review]

**Comment by johnmccabe** [comment]
@johngmyers can you help get these builds setup in Docker Hub

**Review by johnmccabe — APPROVED** [review]

**Pull request merged**